include/extractor/guidance/toolkit.hpp

```diff
@@ -285,6 +288,60 @@ inline std::string applyAccessTokens(std::string lane_string, const std::string
     return result_string;
 }
 
+LaneID inline numLanesToTheRight(const engine::guidance::RouteStep &step)
+{
+    return step.intersections.front().lanes.first_lane_from_the_right;
+}
+
+LaneID inline numLanesToTheLeft(const engine::guidance::RouteStep &step)
+{
+    LaneID total = step.intersections.front().lane_description.size();
+    return total - (step.intersections.front().lanes.lanes_in_turn +
+                    step.intersections.front().lanes.first_lane_from_the_right);
+}
+
+template <class Iterator>
+inline boost::iterator_range<Iterator>
+lanesToTheLeft(const util::guidance::LaneTuple &lanes,
+               const extractor::guidance::TurnLaneDescription &lane_description)
```

Comment:
What about changing the function to
- operate on a `const RouteStep &` similarly to the functions above
- not having to say `lanesToTheLeft<Iterator>(lanes, desc)`

the return type depens on `lane_description`, so you should be able to say

    auto lanes(.., desc) -> iterator_range<decltype(desc.begin())> { make_it_range(..); }

or even

    inline auto lanes(..) { return make_it_range(..); }

then users don't have to care about the iterator type and can simply call the function with the compiler deriving the correct type.